### PR TITLE
401630 - Changes to reduce complextiy for SQ10.4

### PR DESCRIPTION
**Pull request by AtosSK**

### CODE REVIEW

File: src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs [hunk 1]
```diff
@@ -16,12 +16,12 @@ public ConsistentDataSubmissionPeriodsGroupedValidator(IMapper mapper, IIssueCou
...
     }
 
-    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows, List<ProducerValidationEventIssueRequest>? warningRows = null)
+    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows = null, List<ProducerValidationEventIssueRequest>? warningRows = null)
     {
         var remainingErrorCount = await _issueCountService.GetRemainingIssueCapacityAsync(storeKey);
 
-        var firstProducerRow = producerRows.First();
-        var inconsistentDataSubmissionPeriodRow = producerRows.FirstOrDefault(x => x.DataSubmissionPeriod != firstProducerRow.DataSubmissionPeriod);
+        var firstProducerRow = producerRows[0];
```
Comment (Mazar-Shaik) [inline src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs:23]: I think we need to handle null here. Null check?

Comment (AtosSK) [reply src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs:28]: Done.

**Review by Mazar-Shaik — DISMISSED** [review]

**Review by EvidenPete — APPROVED** [review]

File: src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs [hunk 2]
```diff
@@ -16,12 +16,17 @@ public ConsistentDataSubmissionPeriodsGroupedValidator(IMapper mapper, IIssueCou
...
     }
 
-    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows, List<ProducerValidationEventIssueRequest>? warningRows = null)
+    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows = null, List<ProducerValidationEventIssueRequest>? warningRows = null)
     {
         var remainingErrorCount = await _issueCountService.GetRemainingIssueCapacityAsync(storeKey);
 
-        var firstProducerRow = producerRows.First();
-        var inconsistentDataSubmissionPeriodRow = producerRows.FirstOrDefault(x => x.DataSubmissionPeriod != firstProducerRow.DataSubmissionPeriod);
+        if (producerRows == null || producerRows.Count == 0)
```
Comment (hardik-desai-kcl) [inline src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs:23]: could you please write 2 unit test on **old code** i.e. 

var firstProducerRow = producerRows.First();
        var inconsistentDataSubmissionPeriodRow = producerRows.FirstOrDefault(x => x.DataSubmissionPeriod != firstProducerRow.DataSubmissionPeriod);

test 1: producerRows is null
Test 2 producerRows is empty 
and then do your code change and re-run the tests and see what happens?

File: src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/WarningValidators/SinglePackagingMaterialGroupedValidator.cs [hunk 3]
```diff
@@ -48,7 +48,7 @@ public override async Task ValidateAsync(List<ProducerRow> producerRows, string
             }
 
             // Two First() methods called as we need the first item of the first group of data (only one group exists)
-            var representativeRow = distinctPackagingMaterialRows.First().First();
+            var representativeRow = distinctPackagingMaterialRows[0].First();
```
Comment (hardik-desai-kcl) [inline src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/WarningValidators/SinglePackagingMaterialGroupedValidator.cs:51]: could you please write 2 unit test on **old code** i.e.

var representativeRow = distinctPackagingMaterialRows.First().First();
var packagingMaterialSubtypes = distinctPackagingMaterialRows
                .First()
                .GroupBy(row => row.MaterialSubType)
                .ToList();
                
test 1: distinctPackagingMaterialRows.First() is null
Test 2 distinctPackagingMaterialRows.First() is empty
and then do your code change and re-run the tests and see what happens?

File: src/EPR.ProducerContentValidation.Application/Validators/PropertyValidators/DataSubmissionPeriodValidator.cs [hunk 4]
```diff
@@ -49,6 +45,19 @@ public DataSubmissionPeriodValidator()
             .When((row, context) => row.DataSubmissionPeriod != null && DataSubmissionPeriodExists(row.DataSubmissionPeriod, context));
     }
 
+    private static SubmissionPeriodOption GetSubmissionPeriodOption(ValidationContext<ProducerRow> context, ProducerRow row, string dataSubmissionPeriod)
```
Comment (hardik-desai-kcl) [inline src/EPR.ProducerContentValidation.Application/Validators/PropertyValidators/DataSubmissionPeriodValidator.cs:48]: Please add unit test

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

File: src/EPR.ProducerContentValidation.Application/Validators/PropertyValidators/DataSubmissionPeriodValidator.cs (hunk 4 above)
Comment (AtosSK) [reply src/EPR.ProducerContentValidation.Application/Validators/PropertyValidators/DataSubmissionPeriodValidator.cs:48]: This is a private method.

File: src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs (hunk 2 above)
Comment (AtosSK) [reply src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs:23]: Added another test. I had a previous one to check this.

File: sonar-scanner.bat
Comment (hardik-desai-kcl) [inline sonar-scanner.bat:1]: image
Quality Gate failed on this PR, please resolve as required

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

**Review by hardik-desai-kcl — DISMISSED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Review by RaveendraAttota — APPROVED** [review]

**Review by EvidenPete — APPROVED** [review]

**Pull request merged**